### PR TITLE
Fix unwanted ringing of other devices even though the user is already connected to the call.

### PR DESCRIPTION
**Pull request by toger5**
If a call notify event is received but a user is already part of the call (with any other device) no device should ring.
The spec asks clients to never send notify events for already ongoing calls. But since it is possible (and currently happens when EX has not yet synced state when they join a call) it is best to check if the user is connected to the call before ringing.

This is how its expected to behave in the msc text:

> - Current user is a member of the the **call** (room state):\
>   None of the devices should ring if they receive a `m.call.notify` if the
>   rooms state `m.call.member` event of the user contains a membership for
>   the **call** in the `m.call.notify` event.
>   This includes stopping the current ring sound if the room state updates so
>   this condition is true.

https://github.com/matrix-org/matrix-spec-proposals/pull/4075

The updating was already implemented  but the check on notify event retrival was missing.

## Checklist

-   [x] Tests written for new code (and old code if feasible).
-   [x] New or updated `public`/`exported` symbols have accurate [TSDoc](https://tsdoc.org/) documentation.
-   [x] Linter and other CI checks pass.
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md)).

### CODE REVIEW

**Review by t3chguy — CHANGES_REQUESTED** [review]
Please provide a description so someone is able to review it without context

**Comment by t3chguy** [comment]
> Check if the user is already connected to the call.

This doesn't make for a very clear changelog entry, please update

File: src/Notifier.ts [hunk 1]
```diff
@@ -32,6 +32,8 @@ import {
 } from "matrix-js-sdk/src/matrix";
 import { logger } from "matrix-js-sdk/src/logger";
 import { PermissionChanged as PermissionChangedEvent } from "@matrix-org/analytics-events/types/typescript/PermissionChanged";
+// eslint-disable-next-line no-restricted-imports
+import { MatrixRTCSession } from "matrix-js-sdk/src/matrixrtc/MatrixRTCSession";
```
Comment (t3chguy) [inline src/Notifier.ts:36]: The linter is there for a reason, why are we going around it?

File: test/Notifier-test.ts [hunk 2]
```diff
@@ -26,6 +26,10 @@ import {
     SyncState,
 } from "matrix-js-sdk/src/matrix";
 import { waitFor } from "@testing-library/react";
+// eslint-disable-next-line no-restricted-imports
+import { MatrixRTCSession } from "matrix-js-sdk/src/matrixrtc/MatrixRTCSession";
+// eslint-disable-next-line no-restricted-imports
+import { CallMembership } from "matrix-js-sdk/src/matrixrtc/CallMembership";
```
Comment (t3chguy) [inline test/Notifier-test.ts:32]: Ditto. Seems like `matrixrtc` should have an index with all its public exports and we should allowlist that and consume that

Comment (toger5) [reply test/Notifier-test.ts:32]: Thanks for making me finally take care of this...
https://github.com/matrix-org/matrix-js-sdk/pull/4314
and
https://github.com/matrix-org/matrix-react-sdk/pull/12780

I dont think this fits into this PR and I would like to not block this PR on the js-sdk fix (its rather annoying to not have the ringing stop)

So my proposal would be to merge this without the lint rules and I keep the other PR's ontop of this so we can review and merge them independently?

**Review by t3chguy — APPROVED** [review]

**Pull request merged**